pkg/agent/cniserver/server.go

```diff
@@ -492,6 +509,18 @@ func (s *CNIServer) CmdDel(_ context.Context, request *cnipb.CniCmdRequest) (
 		return s.configInterfaceFailureResponse(err), nil
 	}
 	klog.Infof("CmdDel for container %v succeeded", cniConfig.ContainerId)
+
+	if s.secondaryNetworkEnabled {
+		podName := string(cniConfig.K8S_POD_NAME)
+		podNamespace := string(cniConfig.K8S_POD_NAMESPACE)
+
+		containerInfo := s.podConfigurator.podInfoStore.GetCNIConfigInfoByContaninerID(podName, podNamespace, cniConfig.ContainerId)
+		if containerInfo != nil {
+			// Update Virtual Network configuration (retrieved from network-attach-definition) at CNI cache. This is required to release IP address at whereabouts DB.
+			containerInfo.PodCNIStatus = podDelComplete
```

Comment:
It is not safe to do all these changes without locking.

1. containerInfo is a pointer, so not safe to change its field without locking.
2. Between Get and Update, the entry might be removed by podwatch.
3. Another Update is not needed as you already changed the struct field.

Just for this one we may add a func in podInfoStore, like UpdateCNIStatusByContainerID() which updates the status under locking.
However, I guess you might update the Status in podwatch too which needs locking too, and so we should consider that together to design the interfaces.